src/viam/trajex/totg/trajectory.cpp

```diff
@@ -370,66 +378,104 @@ enum class integration_event : std::uint8_t {
+        //  as we we call it in the code q_double_prime(s) is discontinuous. For a path
+        //  generated as described in Section IV the discontinuities are exactly the points
+        //  where the path switches between a circular segment and a straight line segment.
+        //  [However,] If the path's curvature f''(s) is continuous everywhere, this case of
+        //  switching points does not happen."
+        // Reference: Equation 38.
+
         const arc_length boundary = current_segment.end();
 
-        // If this boundary is at the path end, no more interior boundaries to check
+        // If at path end, it is impossible to compare against a subsequent segment since there is none
         if (boundary >= cursor.path().length()) {
             break;
         }
 
-        // Sample geometry from segment ending at boundary
+        // Sample geometry on LEFT side of boundary
         const auto q_prime_before = current_segment.tangent(boundary);
         const auto q_double_prime_before = current_segment.curvature(boundary);
 
-        // Advance cursor to boundary (moves to next segment)
+        // Move cursor to next segment
         cursor.seek(boundary);
         auto segment_after = *cursor;
 
-        // Sample geometry from segment starting at boundary
+        // Sample geometry on RIGHT side of boundary
         const auto q_prime_after = segment_after.tangent(boundary);
         const auto q_double_prime_after = segment_after.curvature(boundary);
 
-        // Compute acceleration limit curve (s_dot_max_acc) on both sides of boundary
-        const auto [s_dot_max_acc_before, s_dot_max_vel_before] =
+        // Compute s_dot_max_acc on both sides
+        const auto [s_dot_max_acc_before, _1] =
             compute_velocity_limits(q_prime_before, q_double_prime_before, opt.max_velocity, opt.max_acceleration, opt.epsilon);
 
-        const auto [s_dot_max_acc_after, s_dot_max_vel_after] =
+        const auto [s_dot_max_acc_after, _2] =
             compute_velocity_limits(q_prime_after, q_double_prime_after, opt.max_velocity, opt.max_acceleration, opt.epsilon);
 
-        // Switching velocity is the minimum of acceleration limit curve on both sides (equation 38)
-        const double s_dot_switching = std::min(s_dot_max_acc_before, s_dot_max_acc_after);
+        // Compute limit curve slopes using numerical approximation
+        const arc_length before_boundary = std::max(boundary - arc_length{opt.epsilon}, current_segment.start());
+        const auto q_prime_bb = current_segment.tangent(before_boundary);
+        const auto q_double_prime_bb = current_segment.curvature(before_boundary);
+        const auto [s_dot_max_acc_bb, _5] =
+            compute_velocity_limits(q_prime_bb, q_double_prime_bb, opt.max_velocity, opt.max_acceleration, opt.epsilon);
+
+        const double actual_step_left = (boundary - before_boundary).value;
+        if (actual_step_left < opt.epsilon * 0.5) {
```

Comment:
What is this check preventing, and why is `continuing` on with the search the right thing to do?

Reply:
`actual_step_left` is used in
```
const double slope_left = (s_dot_max_acc_before - s_dot_max_acc_bb) / actual_step_left;
```

this is preventing us from dividing by a really small number.